Improves algorithm and documentation for term category

hlapp opened the pull request.

johnbradley commented:
When running the tests locally the examples for `term_category` are failing:
```
 > phens <- get_phenotypes("basihyal bone")
  > term_category(phens$id[1:3])
  Error: C stack usage  7970608 is too close to the limit
```
I'm betting we have infinite recursion between `term_category()` and  `as.terminfo()`.

hlapp commented:
> I'm betting we have infinite recursion between `term_category()` and `as.terminfo()`.

I agree recursing over subclasses is arguably a bad idea (as this with near certainty broadens the fan with each recursion). I removed subclass recursion and added a new step for recognition of entities. Seems to work so far.

johnbradley reviewed (approved):
LGTM.
All tests pass when running locally for me now.
